test/testother.cpp

```diff
@@ -3904,6 +3907,127 @@ class TestOther : public TestFixture {
         ASSERT_EQUALS("", errout.str());
     }
 
+    void duplicateExpression7() {
+        check("void f() { const int i = sizeof(int);  if ( i != sizeof (int)){}}\n");
```

Comment:
Can you please add newlines here. I can't see what statements the warning complains about. The assignment is critical so the warning should point out the assignment.